internal/service/common/utils/repository.go

```diff
@@ -9,96 +9,212 @@ import (
+	"github.com/stephenafamo/bob/dialect/psql/im"
 	"github.com/stephenafamo/bob/dialect/psql/sm"
+	"github.com/stephenafamo/bob/dialect/psql/um"
 
 	"github.com/openshift-kni/oran-o2ims/internal/service/common/db"
 )
 
-// Find retrieves a specific tuple from the database table specified.  If no record is found an empty array is returned.
-func Find[T db.Model](ctx context.Context, db *pgxpool.Pool, uuid uuid.UUID, columns []any) ([]T, error) {
+// ErrNotFound represents the error returned by any repository when not record matches the requested criteria
+var ErrNotFound = errors.New("record not found")
+
+// Find retrieves a specific tuple from the database table specified.  If no record is found ErrNotFound is returned
+// as an error; otherwise a pointer to the stored record is returned or a generic error on failure.
+func Find[T db.Model](ctx context.Context, db *pgxpool.Pool, uuid uuid.UUID, columns []any) (*T, error) {
 	// Build sql query
 	var record T
 	if columns == nil {
-		tags := GetAllDBTagsFromStruct(record)
+		tags := GetAllDBTagsFromStruct(record, IncludeNilValues)
 		columns = tags.Columns()
 	}
 
-	query, args, err := psql.Select(
+	sql, args, err := psql.Select(
 		sm.Columns(columns...),
 		sm.From(record.TableName()),
 		sm.Where(psql.Quote(record.PrimaryKey()).EQ(psql.Arg(uuid))),
 	).Build()
 	if err != nil {
-		return []T{}, fmt.Errorf("failed to build query: %w", err)
+		return nil, fmt.Errorf("failed to build query: %w", err)
 	}
 
+	slog.Debug("executing statement", "sql", sql, "args", args)
+
 	// Run query
-	rows, _ := db.Query(ctx, query, args...) // note: err is passed on to Collect* func so we can ignore this
+	rows, _ := db.Query(ctx, sql, args...) // note: err is passed on to Collect* func so we can ignore this
 	record, err = pgx.CollectExactlyOneRow(rows, pgx.RowToStructByNameLax[T])
 	if err != nil {
 		if errors.Is(err, pgx.ErrNoRows) {
-			slog.Info("No Entity found", "uuid", uuid, "table", record.TableName())
-			return []T{}, nil
+			slog.Debug("no record found", "uuid", uuid, "table", record.TableName())
+			return nil, ErrNotFound
 		}
-		return []T{}, fmt.Errorf("failed to call database: %w", err)
+		return nil, fmt.Errorf("failed to execute query: %w", err)
 	}
 
-	slog.Info("record found", "table", record.TableName(), "uuid", uuid)
-	return []T{record}, nil
+	slog.Debug("record found", "table", record.TableName(), "uuid", uuid)
+	return &record, nil
 }
 
 // FindAll retrieves all tuples from the database table specified.  If no records are found then an empty array is
 // returned.
 func FindAll[T db.Model](ctx context.Context, db *pgxpool.Pool, columns []any) ([]T, error) {
+	var record T
+	e := psql.Quote(record.PrimaryKey()).IsNotNull()
+	return Search[T](ctx, db, e, columns)
+}
+
+// Delete deletes a specific tuple from the database table specified.  If no matching tuples are found an error will be
+// returned therefore the caller is responsible for checking for existing records.
```

Comment:
this comment is not ok - I'm fixing it in my PR, so we can leave it as it is for now